cli/ballerina-cli/src/main/java/io/ballerina/cli/cmd/PullCommand.java

```diff
@@ -206,6 +215,35 @@ public void execute() {
         }
     }
 
+    private void pullDependencyPackages(String orgName, String packageName, String version) {
+        Path ballerinaUserHomeDirPath = ProjectUtils.createAndGetHomeReposPath();
+        Path centralRepositoryDirPath = ballerinaUserHomeDirPath.resolve(ProjectConstants.REPOSITORIES_DIR)
+                .resolve(ProjectConstants.CENTRAL_REPOSITORY_CACHE_NAME);
+        Path packagePathInBalaCache = centralRepositoryDirPath.resolve(ProjectConstants.BALA_DIR_NAME)
+                .resolve(orgName).resolve(packageName);
+        Path balaPath = packagePathInBalaCache.resolve(version);
+        String ballerinaShortVersion = RepoUtils.getBallerinaShortVersion();
+        Path cacheDir = centralRepositoryDirPath.resolve(
+                ProjectConstants.CACHES_DIR_NAME + "-" + ballerinaShortVersion);
+
+        ProjectEnvironmentBuilder defaultBuilder = ProjectEnvironmentBuilder.getDefaultBuilder();
+        defaultBuilder.addCompilationCacheFactory(new FileSystemCache.FileSystemCacheFactory(cacheDir));
+        BalaProject balaProject = BalaProject.loadProject(defaultBuilder, balaPath);
+
+        // getResolution pulls all dependencies of the pulled package
+        PackageCompilation packageCompilation = balaProject.currentPackage().getCompilation();
+        DiagnosticResult diagnosticResult = packageCompilation.getResolution().diagnosticResult();
+        if (diagnosticResult.hasErrors()) {
+            for (Diagnostic diagnostic : diagnosticResult.diagnostics()) {
+                CommandUtil.printError(this.errStream, diagnostic.toString(), null, false);
+            }
+            CommandUtil.printError(this.errStream, "package dependency resolution contains errors", null, false);
+            CommandUtil.exitError(this.exitWhenFinish);
+            return;
+        }
+        JBallerinaBackend.from(packageCompilation, JvmTarget.JAVA_11);
```

Comment:
Let's check the diagnostics of JBallerinaBackend as well. There can be cache generation failures due to compilation errors.